runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/DataflowOperationContext.java

```diff
@@ -185,41 +181,19 @@ public abstract static class DataflowExecutionState extends ExecutionState {
     private final ProfileScope profileScope;
     private final @Nullable MetricsContainer metricsContainer;
 
-    /** Clock used to either provide real system time or mocked to virtualize time for testing. */
-    private final Clock clock;
-
     public DataflowExecutionState(
         NameContext nameContext,
         String stateName,
         @Nullable String requestingStepName,
         @Nullable Integer inputIndex,
         @Nullable MetricsContainer metricsContainer,
         ProfileScope profileScope) {
-      this(
-          nameContext,
-          stateName,
-          requestingStepName,
-          inputIndex,
-          metricsContainer,
-          profileScope,
-          Clock.SYSTEM);
-    }
-
-    public DataflowExecutionState(
-        NameContext nameContext,
-        String stateName,
-        @Nullable String requestingStepName,
-        @Nullable Integer inputIndex,
-        @Nullable MetricsContainer metricsContainer,
-        ProfileScope profileScope,
-        Clock clock) {
       super(stateName);
```

Comment:
why the change here from this constructor to super and then setting each field?  Maybe a Java thing?

Reply:
Yes, we need to inititalize super class variables and set of variables specific to this class.